src/gui/guiFormSpecMenu.h

```diff
@@ -352,13 +353,13 @@ class GUIFormSpecMenu : public GUIModalMenu
 	video::SColor m_default_tooltip_color;
 
 private:
-	IFormSource        *m_form_src;
-	TextDest           *m_text_dst;
-	std::string         m_last_formname;
-	u16                 m_formspec_version = 1;
-	std::string         m_focused_element = "";
-	JoystickController *m_joystick;
-	bool m_show_debug = false;
+	IFormSource          *m_form_src;
+	TextDest             *m_text_dst;
+	std::string           m_last_formname;
+	u16                   m_formspec_version = 1;
+	Optional<std::string> m_focused_element = nullopt;
```

Comment:
I've barely touched formspecs, but is `set_focus[]` supposed to reset the focus to nothing, as opposed to the element with an empty name? If so, this might break that, no?

Reply:
According to doc, `set_focus[]` is no special case. So it's supposed to set focus to an element with empty name (if existent).
This should be fine.

(The formspec syntax doesn't differentiate "no arguments" `()` and "one argument with empty string" `(ε,)`).

(Btw. AFAIK there's no reason to have more than one `set_focus` element in a formspec.)